java/source/flight.rst

```diff
@@ -0,0 +1,534 @@
+For more detail about Flight please take a look at `Arrow Flight RPC`_.
+
+.. contents::
+
+Simple Key-Value Storage Service with Arrow Flight
+==================================================
+
+We'll implement a service that provides a key-value store for data, using Flight to handle uploads/requests
+and data in memory to store the actual data.
+
+Flight Client and Server
+************************
+
+.. testcode::
+
+    import org.apache.arrow.flight.Action;
+    import org.apache.arrow.flight.AsyncPutListener;
+    import org.apache.arrow.flight.CallStatus;
+    import org.apache.arrow.flight.Criteria;
+    import org.apache.arrow.flight.FlightClient;
+    import org.apache.arrow.flight.FlightDescriptor;
+    import org.apache.arrow.flight.FlightEndpoint;
+    import org.apache.arrow.flight.FlightInfo;
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.FlightStream;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.flight.NoOpFlightProducer;
+    import org.apache.arrow.flight.PutResult;
+    import org.apache.arrow.flight.Result;
+    import org.apache.arrow.flight.Ticket;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorLoader;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.VectorUnloader;
+    import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+
+    import java.io.IOException;
+    import java.nio.charset.StandardCharsets;
+    import java.util.ArrayList;
+    import java.util.Arrays;
+    import java.util.Collections;
+    import java.util.Iterator;
+    import java.util.List;
+    import java.util.concurrent.ConcurrentHashMap;
+
+    class Datasets {
+        private List<ArrowRecordBatch> batches;
+        private Schema schema;
+        private long rows;
+        public Datasets(List<ArrowRecordBatch> batches, Schema schema, long rows) {
+            this.batches = batches;
+            this.schema = schema;
+            this.rows = rows;
+        }
+        public List<ArrowRecordBatch> getbatches() {
+            return batches;
+        }
+        public Schema getSchema() {
+            return schema;
+        }
+        public long getRows() {
+            return rows;
+        }
+    }
+    class CookbookProducer extends NoOpFlightProducer {
+        private RootAllocator allocator;
+        private Location location;
+
+        public CookbookProducer(RootAllocator allocator, Location location) {
+            this.allocator = allocator;
+            this.location = location;
+        }
+
+        ConcurrentHashMap<FlightDescriptor, Datasets> datasets = new ConcurrentHashMap<>();
```

Comment:
Initialize this in the constructor along with all the other fields, and move this field to be in the same block as all the other fields.

Reply:
Changed